packages/cisco_ftd/data_stream/log/manifest.yml

```
@@ -49,6 +49,14 @@ streams:
         type: bool
         multi: false
         default: false
+      - name: consider_private_internal
+        required: true
+        show_user: true
+        title: Consider private networks as internal
+        description: Assumes CIDR ranges `10.0.0.0/8`, `172.16.0.0/12`, and `192.168.0.0/16` are internal networks.
```

If the proposed behaviour is adopted, this documentation should explain that this feature cannot be used in conjunction with `internal_zones` and `external_zones`.
```suggestion
        description: Assumes CIDR ranges `10.0.0.0/8`, `172.16.0.0/12`, and `192.168.0.0/16` are internal networks. Cannot be used in conjunction with Internal and External Zone lists.
```
However, with the logic change I would write.
```suggestion
        description: Assumes CIDR ranges `10.0.0.0/8`, `172.16.0.0/12`, and `192.168.0.0/16` are internal networks. If used in conjunction with Internal and External Zone lists, private CIDR ranges are used as a fallback resolve direction if not resolved by zone. 
```
or something similar.

---

I don't know that this is necessary. If Zones are configured, the script as I wrote it will not run. Neither the zone-based nor my private CIDR method of calculating `network.direction` will interfere with the other.

---

This is true, but the behaviour demonstrates a non-orthogonality between the configuration parameters. This is a common source of user issues. A user is using private domains in their installation and then later they add zones, the user configures this into the integration but did not include the private range they are using (because for whatever reason they didn't read the docs and there are just so many IPs to add) they don't add them to the list. Now they see that machines that were previously considered internal are not anymore. This results in an issue for them and a possible support case for use. It's better to avoid this.